platform/saivpp/vpplib/SwitchStateBaseFdb.cpp

```diff
@@ -639,3 +641,294 @@ void SwitchStateBase::send_fdb_event_notification(
+        set_sw_interface_l2_bridge(hw_ifname,bridge_id, true);
+
+        //Set interface state up
+        interface_set_state(hw_ifname, true);
+    }else {
+        hw_ifname = hwifname;
+
+        //Create bridge and set the l2 port
+        set_sw_interface_l2_bridge(hw_ifname,bridge_id, true);
+
+        //Set the vlan member to bridge and tags rewrite
+        vpp_l2_vtr_op_t vtr_op = L2_VTR_PUSH_1;
+        vpp_vlan_type_t push_dot1q = VLAN_DOT1Q;
+        uint32_t tag1 = (uint32_t)vlan_id;
+        uint32_t tag2 = ~0;
+        set_l2_interface_vlan_tag_rewrite(hw_ifname, tag1, tag2, push_dot1q, vtr_op);
+    }
+
+    return SAI_STATUS_SUCCESS;
+}
+
+sai_status_t SwitchStateBase::removeVlanMember(
+        _In_ sai_object_id_t objectId)
+{
+    SWSS_LOG_ENTER();
+
+    vpp_remove_vlan_member(objectId);
+
+    auto sid = sai_serialize_object_id(objectId);
+
+    CHECK_STATUS(remove_internal(SAI_OBJECT_TYPE_VLAN_MEMBER, sid));
+
+    return SAI_STATUS_SUCCESS;
+}
+
+sai_status_t SwitchStateBase::vpp_remove_vlan_member(
+        _In_ sai_object_id_t vlan_member_oid)
+{
+    SWSS_LOG_ENTER();
+
+    sai_attribute_t attr;
+
+    attr.id = SAI_VLAN_MEMBER_ATTR_VLAN_ID;
+
+    sai_status_t status = get(SAI_OBJECT_TYPE_VLAN_MEMBER, vlan_member_oid, 1, &attr);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("attr SAI_VLAN_MEMBER_ATTR_VLAN_ID is not present");
+
+        return SAI_STATUS_FAILURE;
+    }
+    sai_object_id_t vlan_oid = attr.value.oid;
+
+    sai_object_type_t ot = objectTypeQuery(vlan_oid);
+
+    if (ot != SAI_OBJECT_TYPE_VLAN)
+    {
+        SWSS_LOG_ERROR("attr SAI_VLAN_MEMBER_ATTR_VLAN_ID is not valid");
+        return SAI_STATUS_FAILURE;
+    }
+
+    attr.id = SAI_VLAN_ATTR_VLAN_ID;
+    status = get(SAI_OBJECT_TYPE_VLAN, vlan_oid, 1, &attr);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("attr SAI_VLAN_ATTR_VLAN_ID is not present");
+
+        return SAI_STATUS_FAILURE;
+    }
+    auto vlan_id = attr.value.u16;
+    uint32_t bridge_id = (uint32_t)vlan_id;
+
+    attr.id = SAI_VLAN_MEMBER_ATTR_BRIDGE_PORT_ID;
+    status = get(SAI_OBJECT_TYPE_VLAN_MEMBER, vlan_member_oid, 1, &attr);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("attr SAI_VLAN_MEMBER_ATTR_BRIDGE_PORT_ID is not present");
+        return SAI_STATUS_FAILURE;
+    }
+
+    sai_object_id_t br_port_oid = attr.value.oid;
+
+    ot = objectTypeQuery(br_port_oid);
+    if (ot != SAI_OBJECT_TYPE_BRIDGE_PORT)
+    {
+        SWSS_LOG_ERROR("SAI_VLAN_MEMBER_ATTR_BRIDGE_PORT_ID=%s expected to be BRIDGE PORT but is: %s",
+                sai_serialize_object_id(br_port_oid).c_str(),
+                sai_serialize_object_type(ot).c_str());
+
+        return SAI_STATUS_FAILURE;
+    }
+    auto &all_bridge_ports = m_objectHash.at(SAI_OBJECT_TYPE_BRIDGE_PORT);
+    bool port_id_found = false;
+    const char *hw_ifname = nullptr;
+    sai_object_id_t port_id;
+    for (auto& bp: all_bridge_ports)
```

Comment:
You don;t need to iterate over the all_bridge_ports.
all_bridge_ports is pointing to a hash map. You can use serialized objectid to index into the map

typedef std::map<sai_object_type_t, std::map<std::string, AttrHash>> ObjectHash

Reply:
adressed